dali/core/mm/pool_resource_test.cc

```diff
@@ -202,6 +205,104 @@ TEST(MMPoolResource, DeferredCheckAsync) {
   }
 }
 
+namespace {
+
+struct block {
+  void *ptr;
+  size_t size;
+  uint8_t fill;
+};
+
+template <typename Pool, typename Mutex>
+void PoolTest(Pool &pool, vector<block> &blocks, Mutex &mtx, int max_iters = 20000) {
```

Comment:
This function is a close relative of the one used in `async_pool_test` - just simpler, because there's no stream logic and the memory is on host.